src/bitcoin.cpp

```diff
@@ -0,0 +1,287 @@
+                        escaped.append(backslash_count * 2 + 1, '\\');
+                        escaped.push_back('"');
+                    } else {
+                        // Otherwise, backslashes remain as-is
+                        escaped.append(backslash_count, '\\');
+                        --i; // Compensate for the outer loop's increment
+                    }
+                } else if (arg[i] == '"') {
+                    // Escape double quotes with a backslash
+                    escaped.push_back('\\');
+                    escaped.push_back('"');
+                } else {
+                    escaped.push_back(arg[i]);
+                }
+            }
+            escaped.push_back('"'); // End with a quote
+            escaped_args.emplace_back(std::move(escaped));
+            new_argv.push_back((char *)escaped_args.back().c_str());
+        }
+    }
+    new_argv.push_back(nullptr);
+    return _execvp(file, new_argv.data());
+#endif
+}
+
+//! Execute the specified bitcoind, bitcoin-qt or other command line in `args`
+//! using src, bin and libexec directory paths relative to this executable, where
+//! the path to this executable is specified in `wrapper_argv0`.
+//!
+//! @param args Command line arguments to execute, where first argument should
+//!             be a relative path to a bitcoind, bitcoin-qt or other executable
+//!             that will be located on the PATH or relative to this executable.
+//!
+//! @param wrapper_argv0 String containing first command line argument passed to
+//!                      main() to run the current executable. This is used to
+//!                      help determine the path to the current executable and
+//!                      how to look for new executables.
+//
+//! @note This function doesn't currently print anything but can be debugged
+//! from the command line using strace or dtrace like:
+//!
+//!     strace -e trace=execve -s 10000 build/src/bitcoin ...
+//!     dtrace -n 'proc:::exec-success { trace(curpsinfo->pr_psargs); }' -c ...
+void ExecCommand(const std::vector<const char*>& args, std::string_view wrapper_argv0)
+{
+    // Construct argument string for execvp
+    std::vector<const char*> exec_args{args};
+    exec_args.emplace_back(nullptr);
+
+    // Try to call execvp with given exe path.
+    auto try_exec = [&](fs::path exe_path, bool allow_notfound = true) {
+        std::string exe_path_str{fs::PathToString(exe_path)};
+        exec_args[0] = exe_path_str.c_str();
+        if (ExecVp(exec_args[0], (char*const*)exec_args.data()) == -1) {
+            if (allow_notfound && errno == ENOENT) return false;
+            throw std::system_error(errno, std::system_category(), strprintf("execvp failed to execute '%s'", exec_args[0]));
+        }
+        return true; // Will not actually be reached if execvp succeeds
+    };
+
+    // Try to figure out where wrapper executable is located. This does a
```

Comment:
I wonder if this can be made dumb stupid and reduce it to just call `execvp()`. That is, reduce `ExecCommand()` to just 1 - 5 lines of code and deal with the consequences of it...

Reply:
I don't have any particular suggestion. I will post here if I come up with some significant simplifications.